### PR TITLE
ci: clean up build workflows

### PR DESCRIPTION
**Pull request by MariusBgm**
Move the runner setup into the workflows and keep the build-cmake-preset
action as simple as possible.
Add workflow_dispatch to mingw so we can run it manually.

### CODE REVIEW

File: .github/workflows/build-mingw64.yml [hunk 1]
```diff
@@ -1,60 +1,46 @@
...
+      - name: install packages
         run: |
-          echo "C:\msys64" >> $GITHUB_PATH
-          echo "C:\msys64\mingw64\bin\" >> $GITHUB_PATH
-      - name: Install Dependencies
-        run: |
-          pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja --noconfirm
-          export PATH="/mingw64/bin/:$PATH"
-        shell:  C:\shells\msys2bash.cmd {0}
+          pacman -S mingw-w64-${{ matrix.builds.arch }}-gcc mingw-w64-${{ matrix.builds.arch }}-cmake mingw-w64-${{ matrix.builds.arch }}-ninja --noconfirm
```
Comment (VDanielEdwards) [inline .github/workflows/build-mingw64.yml:33]: This does install `cmake` in the MSYS2 environment, but ...

File: .github/workflows/build-mingw64.yml [hunk 2]
```diff
@@ -1,60 +1,46 @@
...
-        shell:  C:\shells\msys2bash.cmd {0}
-      - name: Test
-        id: cmake-test-mingw64
-        run: |
-          export PATH="/mingw64/bin/:$PATH"
-          echo "PATH: $PATH"
-          ctest --preset gcc-release --output-on-failure
-        shell:  C:\shells\msys2bash.cmd {0}
+
+      - uses: ./.github/actions/build-cmake-preset
```
Comment (VDanielEdwards) [inline .github/workflows/build-mingw64.yml:40]: .. this action does not use the MSYS2 environment for the build. The default shell is not passed to the action, it's only used for the workflow itself.

Comment (MariusBgm) [reply .github/workflows/build-mingw64.yml:40]: ah, thanks. that's the magic sauce that I missed :-)

**Review by VDanielEdwards — APPROVED** [review]
If the `cmake` executable can be found again in the correct MinGW installation, this PR is ready for merging 👍

**Review by VDanielEdwards — APPROVED** [review]

**Comment by MariusBgm** [comment]
> If the `cmake` executable can be found again in the correct MinGW installation, this PR is ready for merging 👍

Yes, the cmake executable is now properly found. 
It helps that we can rely on the fact that there is a bash available on all systems. 
We now export the extra-paths in the bash shell of the composite action. This allows setting the path for the mingw / msys2 installations from outside of the `build-cmake-preset`-action.

**Pull request merged**